plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+			wantErr:     true,
+			expectedErr: `{"config":{"timeout":"value should be between 0 and 2147483646"}}`,
+		},
+		{
+			name: "valid uuid",
+			config: `{
+				"uuid": "23e623b0-ca23-11ec-9d64-0242ac120002",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid uuid",
+			config: `{
+				"uuid": "abc",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a valid UUID"}}`,
+		},
+		{
+			name: "invalid empty uuid",
+			config: `{
+				"uuid": "",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a valid UUID"}}`,
+		},
+		{
+			name: "invalid not-string uuid",
+			config: `{
+				"uuid": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a string"}}`,
+		},
+		{
+			name: "valid auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": 1234,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid number auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": 1.234,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_s":"expected an integer"}}`,
+		},
+		{
+			name: "invalid string auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": "1234",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_s":"expected an integer"}}`,
+		},
+		{
+			name: "valid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": 1.234,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": "1.234",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_ms":"expected a number"}}`,
+		},
+		{
+			name: "valid name",
+			config: `{
+				"name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid name",
+			config: `{
+				"name": "foo.-_~!",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{
+				"name":"invalid value 'foo.-_~!': it must only contain alphanumeric and '., -, _, ~' characters"
+				}}`,
+		},
+		{
+			name: "valid utf8_name",
+			config: `{
+				"utf8_name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid name",
+			config: `{
+				"utf8_name": "foo.-_~!",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{
+				"utf8_name":"invalid value 'foo.-_~!': the only accepted ascii characters are alphanumerics or ., -, _, and ~"
+				}}`,
+		},
+		{
+			name: "valid sni",
+			config: `{
+				"sni": "www.konghq.com",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid sni with port",
+			config: `{
+				"sni": "www.konghq.com:8080",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"sni":"must not have a port"}}`,
+		},
+		{
+			name: "invalid sni as IP",
+			config: `{
+				"sni": "192.0.2.0",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"sni":"must not be an IP"}}`,
+		},
```

Comment:
Since this goes through `utils.normalize_ip` we should also make sure it fails when using an IPv6 address.

Reply:
https://github.com/Kong/goks/pull/14/commits/b889f8cc359429ddcf58518f65cdb983c736b01f